src/modules/content-seeding/news-aggregator.js

```diff
@@ -0,0 +1,290 @@
+import { bus, config } from '../../core-server';
+import winston from 'winston';
+import * as pg from '../../lib/pg';
+import promisify from '../../lib/promisify';
+import uuid from 'node-uuid';
+import { TYPE_THREAD, TAG_RSS_NEWS_SEED } from '../../lib/Constants';
+import type { Thread } from '../../lib/schemaTypes';
+
+/*
+	Implementation Details:
+		- TABLES:
+			- FEEDS:
+				column                   type
+			  -----------------------------------------
+			    url		                text
+		        MTBU                    float(24)
+				lastrequesttime         bigint
+				lastupdatetime          bigint
+
+			- ARTICLES:
+				column                   type
+			  -----------------------------------------
+			    url		                text
+		        rawjson                 jsonb
+				terms    		        tsvector
+
+		- ALGORITHM:
+			- Check if the feeds table is empty.
+			- IF empty:
+				- insert a seed url in the feeds table (one time operation).
+			- Truncate the articles table.
+			- Filter feeds where lastrequesttime < ( NOW() - Mean time between updates (MTBU) ).
+			- For each feeds:
+				- Check if the the given feed has some new news article (RSS FEED'S latest article timestamp > lastupdatetime).
+				- If updated, update MTBU, lastrequesttime, lastupdatetime else update lastrequesttime only.
+						MTBU = Average time interval at which a news story is added on that feed.
+				- make a insert query in articles table with url, articleJson, and ts_vector(article_title + article_content).
+				- select roomid, rawjson, url from rooms and articles table where terms lexemes match the room name.
+				- For each result:
+					- Construct a thread with title, url, description from articles and post it into relevent rooms.
+			- Invoke this job every hour.
+*/
+
+
+const performReadQuery = promisify(pg.read.bind(pg, config.connStr));
+const performWriteQuery = promisify(pg.write.bind(pg, config.connStr));
+const parseFeed = promisify(feedParser.bind(feedParser));
+
+type Article = {
+	title: string;
+	link: string;
+	content: string;
+	published: Date;
+}
+
+type FeedsObject = {
+	url: string;
+	mtbu: number;
+	lastrequesttime: number;
+	lastupdatetime: number
+};
+
+type RoomSpecificNews = {
+	roomid: string;
+	roomname: string;
+	article: Article;
+	url: string
+};
+
+function articleAge (articlePubDate: number): number {
+	return Date.now() - articlePubDate;
+}
+
+function buildThreads (latestNewsForRooms: Array<RoomSpecificNews>): Thread {
+	const changes = {
+		entities: {}
+	};
+	for (const newsArticle of latestNewsForRooms) {
+		const id = uuid.v4();
+		changes.entities[id] = {
+			id,
+			type: TYPE_THREAD,
+			name: newsArticle.article.title,
+			body: newsArticle.article.link,
+			tags: [ TAG_RSS_NEWS_SEED ],
+			parents: [ newsArticle.roomid ],
+			identities: [],
+			creator: 'belongbot',
+			createTime: Date.now(),
+		};
+	}
+	return changes;
+}
+
+function calculateMTBU (newArticles: Array<Article>): number {
```

Comment:
You can merge this with the function that updates MTBU in the database.

This is a lot of work for what is essentially a rough heuristic. I would have simply gone with
```
function updateFeed(
   feed /* current feed object */,
   newArticleCount /* The number of new articles seen in the current pass */
) {
    return write ([{
        $: 'UPDATE feeds SET lastreadtime = &{now}, mtbu = &{mtbu}' +
            (newArticleCount ? ', lastupdatetime = &{now}') +
            ' WHERE url = &{url}',
        now: Date.now(), url: feed.url
        mtbu: (Date.now() - feed.lastupdatetime) / newArticleCount
    }]);
}
```